Source/WebKit/UIProcess/ios/DragDropInteractionState.mm

```diff
@@ -403,15 +402,23 @@ static bool canUpdatePreviewForActiveDragSource(const DragSourceState& source)
             continue;
 
         if (source.action.contains(DragSourceAction::Link)) {
-            dragItem.previewProvider = [title = retainPtr((NSString *)source.linkTitle), url = retainPtr((NSURL *)source.linkURL), center = source.adjustedOrigin] () -> UIDragPreview * {
-                UIURLDragPreviewView *previewView = [UIURLDragPreviewView viewWithTitle:title.get() URL:url.get()];
-                previewView.center = center;
-                auto parameters = adoptNS([[UIDragPreviewParameters alloc] initWithTextLineRects:@[ [NSValue valueWithCGRect:previewView.bounds] ]]);
-                [parameters setBackgroundColor:[UIColor colorWithDynamicProvider:[] (UITraitCollection *traitCollection) -> UIColor * {
+            dragItem.previewProvider = [title = retainPtr((NSString *)source.linkTitle), url = retainPtr((NSURL *)source.linkURL)] () -> UIDragPreview * {
+                RetainPtr preview = [UIDragPreview previewForURL:url.get() title:title.get()];
+#if PLATFORM(VISION)
```

Comment:
As I understand it, this adjustment is only needed on visionOS to fix rdar://101837369. iOS doesn't require this because the URL platter will use an opaque color (respecting system dark mode) by default, but on Vision it's transparent.

(Perhaps @hortont424 remembers the full context?)

Reply:
I see! So it's just to avoid unnecessary work on iOS.